Extension/c_cpp_properties.schema.json

```diff
@@ -107,6 +107,14 @@
                 }
             }
         },
+        "variables" : {
```

Comment:
I debated between "environment" and "variables", but I'm open to suggestions on what to call this.  Visual Studio's CppProperties.json has an array called "environments" (yes, plural) that lets you do fancy namespacing, but i don't think we want that.

Reply:
launch.json and tasks.json use "env", so I'm thinking maybe we'll go with that to be consistent.